src/crimson/osd/osd.cc

```diff
@@ -273,6 +273,18 @@ seastar::future<> OSD::start()
   }).then([this] {
     return heartbeat->start(public_msgr.get_myaddrs(),
                             cluster_msgr.get_myaddrs());
+  }).then([this] {
+    // attach the CCT's AdminSocket object to the OSD's admin socket path
+    auto x = local_conf().get_val<std::string>("admin_socket");
+    return shard_services.get_cct()->get_admin_socket()->init(
+                                                  x.length() ? x : "/tmp/asok");
```

Comment:
Is it possible for x to be null?  It has a default for daemons, so I think you can ceph_assert(x) here.

Reply:
Fixed (I used it for testing, but that shouldn't be in the production code).
Added a check in AdminSocket::init() for the missing config.